src/upload.rs

```diff
@@ -1,3 +1,6 @@
+//! This module handles the expected information a crate should have
+//! and manages the serialising and deserialising of this information
```

Comment:
Oh I was actually surprised to see *serialize* stuff in here, so I investigated a bit, and I'm pretty sure the serialization is only used in tests. I think that'd be good to note in this comment :)